dbms/src/Flash/Statistics/ExecutorStatistics.h

```diff
@@ -115,5 +127,31 @@ class ExecutorStatistics : public ExecutorStatisticsBase
     virtual void appendExtraJson(FmtBuffer &) const {}
 
     virtual void collectExtraRuntimeDetail() {}
+
+    void collectJoinBuildTime()
+    {
+        /// for join need to add the build time
+        /// In TiFlash, a hash join's build side is finished before probe side starts,
+        /// so the join probe side's running time does not include hash table's build time,
+        /// when construct Execution Summaries, we need add the build cost to probe executor
+        auto all_join_id_it = dag_context.getExecutorIdToJoinIdMap().find(executor_id);
+        if (all_join_id_it != dag_context.getExecutorIdToJoinIdMap().end())
+        {
+            for (const auto & join_executor_id : all_join_id_it->second)
+            {
+                auto it = dag_context.getJoinExecuteInfoMap().find(join_executor_id);
+                if (it != dag_context.getJoinExecuteInfoMap().end())
+                {
+                    UInt64 time = 0;
+                    for (const auto & join_build_stream : it->second.join_build_streams)
+                    {
+                        if (auto * p_stream = dynamic_cast<IProfilingBlockInputStream *>(join_build_stream.get()); p_stream)
+                            time = std::max(time, p_stream->getProfileInfo().execution_time);
+                    }
+                    process_time_for_join_build += time;
```

Comment:
sum all join_build's time together might not be very useful for users to analyze performance. It would be nice if record build time info for each hash join.

Reply:
> sum all join_build's time together might not be very useful for users to analyze performance. It would be nice if record build time info for each hash join.

Good idea, maybe I can support it in Pipeline model

Reply:
Yeah, especially, join_builds can run concurrently, sum them up would be mis-understanding.